packages/opencensus-exporter-trace-zipkin/src/zipkin.ts

```diff
@@ -0,0 +1,178 @@
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as http from 'http';
+import * as url from 'url';
+
+export interface ZipkinExporterOptions extends types.ExporterConfig {
+  url: string;
+  serviceName: string;
+}
+
+interface TranslatedSpan {
+  traceId: string;
+  name: string;
+  id: string;
+  parentId?: string;
+  kind: string;
+  timestamp: string;
+  duration: string;
+  debug: boolean;
+  shared: boolean;
+  localEndpoint: {serviceName: string};
+}
+
+/** Zipkin Exporter manager class */
+export class ZipkinTraceExporter implements types.Exporter {
+  private zipkinUrl: url.UrlWithStringQuery;
+  private serviceName: string;
+  buffer: classes.ExporterBuffer;
+  logger: types.Logger;
+
+  constructor(options: ZipkinExporterOptions) {
+    this.zipkinUrl = url.parse(options.url);
+    this.serviceName = options.serviceName;
+    this.buffer = new classes.ExporterBuffer(this, options);
+    this.logger = options.logger || logger.logger();
+  }
+
+  /**
+   * Is called whenever a span is ended.
+   * @param root the ended span
+   */
+  onEndSpan(root: types.RootSpan) {
+    this.buffer.addToBuffer(root);
+  }
+
+  /**
+   * Send a trace to zipkin service
+   * @param zipkinTraces Trace translated to Zipkin Service
+   */
+  private async sendTraces(zipkinTraces: TranslatedSpan[]) {
+    /** Request options */
+    const options = {
+      hostname: this.zipkinUrl.hostname,
+      port: this.zipkinUrl.port,
+      path: this.zipkinUrl.path,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    };
+
+    return new Promise((resolve, reject) => {
+      try {
+        /** Request object */
+        const req = http.request(options, (res) => {
+          res.on('data', (chunk) => {});
+          // Resolve on end
+          res.on('end', () => {
+            resolve(
+                {statusCode: res.statusCode, statusMessage: res.statusMessage});
+          });
+        });
+
+        /** Request error event */
+        req.on('error', (e) => {
+          reject({
+            statusCode: 500,
+            statusMessage: `Problem with request: ${e.message}`
+          });
+        });
+
+        /** Request body */
+        const spansJson: string[] =
+            zipkinTraces.map((span) => JSON.stringify(span));
+        spansJson.join('');
+        const outputJson = `[${spansJson}]`;
+        this.logger.debug('Zipkins span list Json: %s', outputJson);
+
+        // Sendind the request
+        req.write(outputJson, 'utf8');
+        req.end();
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  /**
+   * Mount a list (array) of spans translated to Zipkin format
+   * @param rootSpan Trace to be translated
+   */
+  private mountSpanList(spanList: TranslatedSpan[], rootSpan: types.RootSpan):
```

Comment:
It seems like `spanList` is not really necessary, since you can just construct a new array within this function instead.

Reply:
Done.